.github/workflows/documentation.yml

```diff
@@ -101,17 +142,255 @@ jobs:
         with:
           name: dnsdist-html-docs-${{steps.get-version.outputs.pdns_version}}
           path: ./pdns/dnsdistdist/docs/_build/dnsdist-html-docs.tar
-      - run: bzip2 dnsdist-html-docs.tar
-        if: ${{github.ref_name == 'master'}}
-        working-directory: ./pdns/dnsdistdist/docs/_build
       - run: inv ci-docs-build-pdf
         working-directory: ./pdns/dnsdistdist
       - uses: actions/upload-artifact@v4
         with:
           name: dnsdist-${{steps.get-version.outputs.pdns_version}}.pdf
           path: ./pdns/dnsdistdist/docs/_build/latex/dnsdist.pdf
-      - run: inv ci-docs-upload-master --docs-host="${DOCS_HOST}" --pdf="dnsdist.pdf" --username="dnsdist_org" --product="dnsdist"
+      - run: docker logs ${{ steps.proxpi-docker.outputs.id }} 2>&1 | grep whl | awk '{print $8}' | cut -d "/" -f 4 | awk -F'-' '{print $1"=="$2}' | sort -u --ignore-case > /tmp/proxpi-dnsdist.log
+      - id: pip-logs-dnsdist
+        run: echo "list-dnsdist=$(cat /tmp/proxpi-dnsdist.log | base64 -w0)" >> "$GITHUB_OUTPUT"
+      - name: Copy PDF to HTML docs directory
+        run: |
+          cp ./pdns/dnsdistdist/docs/_build/latex/dnsdist.pdf ./pdns/dnsdistdist/docs/_build/dnsdist-html-docs/
+
+      # Build website
+      - name: Build website
+        run: |
+          tar cf website.tar website
+      - uses: actions/upload-artifact@v4
+        with:
+          name: website-${{steps.get-version.outputs.pdns_version}}
+          path: website.tar
+
+  publish-to-netlify:
+    name: Publish to Netlify
+    needs: build-docs
+    if: ${{ vars.PUBLISH_DOCS_TO_NETLIFY == 'true' && (github.ref_name == 'master' || vars.PUBLISH_PRS_TO_NETLIFY == 'true') }}
+    runs-on: ubuntu-22.04
+    env:
+      NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_API_TOKEN }}
```

Comment:
because PRs run untrusted code and, by default, don't have access to secrets (and honestly we wouldn't want them to!)

Reply:
Oh, interesting! I didn't know that. Maybe we just ditch the Netlify PR thing, or the Netlify thing in general, then? Or do you see any other way?

Reply:
I do know that https://github.com/esphome/esphome-devices has netlify PR previews, with no mention of netlify anywhere in their workflows. But maybe we skip the PR thing now so we can carry on.

Reply:
Instead of pushing to Netlify you can also configure it to watch your repo, but then you have to build the docs on their infrastructure, which would mean porting all this to stick it into a container and dealing with their restrictions on resource usage potentially. I'll remove the PR thing for now and we can revisit this. For the moment it's mostly a nice to have, I think :)

Reply:
Generated a tarball containing the generated website files and uploading it as an artifact would be nice